app/vmctl/vm_native.go

```diff
@@ -262,7 +263,7 @@ func (p *vmNativeProcessor) runBackfilling(ctx context.Context, tenantID string,
 
 	// any error breaks the import
 	for mName, mRanges := range metrics {
-		match, err := buildMatchWithFilter(p.filter.Match, mName)
+		match, err := buildMatchWithFilter(p.filter.Match, "!=", mName)
```

Comment:
if we use `!=` as operator than we will build name filter like `{__name__!="metric_name"}` and it will make incorrect filter requests. By default, we should get the metrics by its name `{__name__="metrics_name"}`.

Reply:
updated

Reply:
I have checked the process and it works incorrectly. If I use filter like in the example below
```
2024/10/22 10:44:28 Initing import process from "http://localhost:8481/select/0:0/prometheus/api/v1/export/native" to "http://localhost:8480/insert/0:0/prometheus/api/v1/import/native" with filter 
        filter: match[]={__name__=~"vm_.*"}
        start: 2024-10-10T14:55:00Z for tenant 0:0
2024/10/22 10:44:28 Selected time range will be split into 282 ranges according to "hour" step
2024/10/22 10:44:28 
Total: 0 ↖ Speed: ? p/s                                                                                                                                                                                             Continue import process with filter 
        filter: match[]={__name__!=""}
        start: 2024-10-10T15:55:00Z
        end: 2024-10-10T16:55:00Z:
Total: 8.42 MiB ↘ Speed: 1.41 MiB p/s

```
if starts to fetch all metrics instead of only by defined filter

Reply:
removed metricName filter if metricName is empty